packages/core/src/api/nodeConversions/nodeConversions.ts

```diff
@@ -349,11 +350,14 @@ function contentNodeToInlineContent(contentNode: Node) {
 /**
  * Convert a TipTap node to a BlockNote block.
  */
-export function nodeToBlock<BSchema extends BlockSchema>(
+export function nodeToBlock<
+  BSchema extends BlockSchema,
+  BType extends keyof BSchema
```

Comment:
BType now functions as an "unsafe" cast. I'm not sure that's desirable, because we don't know for sure whether the node is of this type, right?

Should we pass a parameter with `blockType: BType`, so we can do a run-time check on whether it's a matching node?

Reply:
I think that shouldn't be necessary. The idea here is that you can get `BType` from `node.type.name`, and provided that the node came from an editor with the same `blockSchema` as you passed to `nodeToBlock`, you should end up with a `Block` of type `BType`. Unless there's a scenario where the node comes from a different editor to the schema, I think it should be ok.

There is one caveat, being that TypeScript has no way to infer `BType` from `node.type.name` since `Node` is not that strictly typed. So you have to provide the type arguments yourself if you need them.